src/LegacyEventMapper.php

```diff
@@ -183,6 +192,20 @@ private function ceResourceAndSubject(string $ceService, string $resource): arra
                 $ret === 0 ? 'Resource regex did not match' : 'Failed while matching resource regex'
             );
         }
+        if ('firebasedatabase.googleapis.com' === $ceService) {
+            if (null === $domain) {
+                return [null, null];
+            }
+            $location = 'us-central1';
+            if ($domain !== 'firebaseio.com') {
+                preg_match('#^([\w-]+)\.firebasedatabase\.app$#', $domain, $locationMatches);
```

Comment:
We are actually more lenient than this in other languages as we aren't sure that `firebasedatabase.app` will be the only domain possible. You can basically match everything up to the first dot. 

[We weakened the regex in the Ruby FF after the original PR I pointed you to](https://github.com/GoogleCloudPlatform/functions-framework-ruby/pull/109/files)

Reply:
@matthewrobertson done in 42ae03dadb77227d5bfef17f1cf6999ef235a04d